state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1670,6 +1650,51 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 	}
 
+	fn try_broadcast_rejection_refund_or_store_tx_details(
+		tx: TransactionRejectionDetails<T, I>,
+		refund_address: <T::TargetChain as Chain>::ChainAccount,
+		deposit_fetch_id: Option<<T::TargetChain as Chain>::DepositFetchId>,
+	) {
+		let AmountAndFeesWithheld {
+			amount_after_fees: amount_after_ingress_fees,
+			fees_withheld: _,
+		} = Self::withhold_ingress_or_egress_fee(
+			IngressOrEgress::IngressDepositChannel,
```

Comment:
The fee is different depending on whether it's a vault ingress or deposit channel ingress. (see `conditionally_withhold_ingress_fee`).

Reply:
Maybe this would work:

```
if deposit_fetch_id.is_some() {
    IngressOrEgress::IngressDepositChannel
} else {
    IngressOrEgress::IngressVaultSwap
}
```